src/components/config_profile/include/config_profile/profile.h

```diff
@@ -230,7 +235,7 @@ class Profile : public protocol_handler::ProtocolHandlerSettings,
   /**
    * @brief Returns desirable thread stack size
    */
-  const uint64_t& thread_min_stack_size() const;
+  const uint64_t thread_min_stack_size() const;
```

Comment:
This changes the method signature, we have started to be stricter on semantic versioning for any public functionality. As such, the old signature should be deprecated, but left until a major version update.

Reply:
This goes the same for any method signatures that were changed in this PR.

Reply:
@jacobkeeler return value is not exactly part of function signature. I agree that is it important, but in that case redundant reference to POD type was removed, and I think that it is not major changes of method API. Also note, that we are not able to mark it deprecated and create another one method without reference in return value, it would be not compilable